test/Test/JsOutput.hs

```diff
@@ -100,6 +100,101 @@ suite =
             expectTextContains jsOutput "n1 = 0"
             expectTextContains jsOutput "n2 = $author$project$Main$N2$(0, 0)"
 
+        Nothing ->
+          crash "JS output could not be read."
+    , scope "direct function calls - mutual recursion" $ do
+      project <- io $ Lamdera.Relative.requireDir "test/direct-fn-calls-mutual-recursion"
+      let
+        elmHome = project ++ "/elm-home"
+        elmStuff = project ++ "/elm-stuff"
+
+      maybeJsOutput <- io $ do
+        rmdir elmHome
+        rmdir elmStuff
+
+        Test.Helpers.withElmHome elmHome $
+          Ext.Common.withProjectRoot project $
+            Make.run ["src/Main.elm"] $
+              Make.Flags
+                { _debug = False
+                , _optimize = True
+                , _output = Just (Make.JS "elm-stuff/tmp.js")
+                , _report = Nothing
+                , _docs = Nothing
+                , _noWire = True
+                , _optimizeLegible = False
+                }
+
+        fileContents <- readUtf8Text $ elmStuff ++ "/tmp.js"
+
+        rmdir elmHome
+        rmdir elmStuff
+
+        pure fileContents
+
+      case maybeJsOutput of
+        Just jsOutput ->
+          do
+            expectTextContains jsOutput "$Main$a2 = function (n) {"
+            expectTextContains jsOutput "$Main$cyclic$a1() {"
+            expectTextContains jsOutput "$Main$a1 ="
+            expectTextContains jsOutput "$Main$cyclic$a1 = function () {"
+            expectTextContains jsOutput "$Main$a1(1)"
+            expectTextContains jsOutput "$Main$a2(1)"
+
+            expectTextContains jsOutput "$Main$b2$ = function (m, n) {"
+            expectTextContains jsOutput "$Main$cyclic$b1()"
+            expectTextContains jsOutput "$Main$b2 = F2("
+            expectTextContains jsOutput "$Main$cyclic$b1 = "
+            expectTextContains jsOutput "$Main$b1 ="
+            expectTextContains jsOutput "$Main$cyclic$b1 = function () {"
+            expectTextContains jsOutput "$Main$b1, 1, 1)" -- A2
+            expectTextContains jsOutput "$Main$b2$(1, 1)"
```

Comment:
With the test we're showing that the designated "beginning of the cycle" doesn't get called directly (there's no `$b1$`), but the other parts of the cycle do.